libs/media-player/src/lib/audio-clip-player.tsx

```diff
@@ -0,0 +1,101 @@
+import { isNullOrUndefined } from '@coscrad/validation-constraints';
+import BugReportIcon from '@mui/icons-material/BugReport';
+import LinkOffIcon from '@mui/icons-material/LinkOff';
+import { Box, CircularProgress, IconButton, Tooltip } from '@mui/material';
+import { useEffect, useState } from 'react';
+import { DefaultPlayButton } from './default-clip-player-button';
+
+interface IPlayButtonProps {
+    onButtonClick: () => void;
+}
+
+export interface IPlayButton {
+    (props: IPlayButtonProps): JSX.Element;
+}
+
+export interface AudioClipPlayerProps {
+    audioUrl: string;
+    UserDefinedPlayButton?: IPlayButton;
+}
+
+enum AudioState {
+    loading = 'loading',
+    error = 'error',
+    canPlay = 'canPlay',
+    undefined = 'undefined',
```

Comment:
Can we rename this? It's not that the audio is in an `undefined` state . call it
```ts
isMissingAudioLink = 'isMissingAudioLink'
```